### PR TITLE
Fix navigator 2.0 in Flutter Web

### PR DESCRIPTION
**Pull request by jacobaraujo7**
## Description

In Flutter Web the page is expected to notify delegates when there is a change in the URL. What did not happen in the BETA version of Flutter, only worked in STABLE.

What happened was that when we are using nullsafety we will have an error when trying to cast a null value, as it happened in the line 663 **(bindings.dart)**

```dart
Future<void> _handlePushRouteInformation(Map<dynamic, dynamic> routeArguments) async {
    for (final WidgetsBindingObserver observer in List<WidgetsBindingObserver>.from(_observers)) {
      if (
        await observer.didPushRouteInformation(
          RouteInformation(
            location: routeArguments['location'] as String,
            state: routeArguments['state'] as Object,
          )
        )
      )
      return;
    }
  }
```

The variable routeArguments['state'] is always  null in Flutter Web and the attempt to cast caused this error:
Screen Shot 2020-12-12 at 15 40 07

The solution is very simple, just change that (as Object) to that (as Object?) Getting like this:

```dart
Future<void> _handlePushRouteInformation(Map<dynamic, dynamic> routeArguments) async {
    for (final WidgetsBindingObserver observer in List<WidgetsBindingObserver>.from(_observers)) {
      if (
        await observer.didPushRouteInformation(
          RouteInformation(
            location: routeArguments['location'] as String,
            state: routeArguments['state'] as Object?,
          )
        )
      )
      return;
    }
  }
```

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jacobaraujo7** [comment]
@chunhtai look this

**Review by chunhtai — COMMENTED** [review]
thanks for the fix, can you add a test to make sure we don't accidentally break it?

**Comment by jacobaraujo7** [comment]
@googlebot I fixed it.

**Comment by jacobaraujo7** [comment]
> thanks for the fix, can you add a test to make sure we don't accidentally break it?

Done! @chunhtai

**Comment by chunhtai** [comment]
Hi @jacobaraujo7 , We may need you to sign the CLA before we can merge this PR

**Comment by jacobaraujo7** [comment]
@googlebot I fixed it.

**Comment by jacobaraujo7** [comment]
@googlebot I fixed it.

**Comment by jacobaraujo7** [comment]
> Oi @jacobaraujo7 , Podemos precisar de você para assinar o CLA antes que possamos fundir este RP

@chunhtai 
image

I have already signed this contract. There is no author besides me

**Comment by jacobaraujo7** [comment]
@googlebot I fixed it.

**Comment by chunhtai** [comment]
I think this commit is the culprit
Screen Shot 2020-12-17 at 10 00 35 AM

Can you try squash all commits and make sure you have the right email?

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by jacobaraujo7** [comment]
@googlebot I fixed it.

**Comment by jacobaraujo7** [comment]
@googlebot I fixed it.

**Comment by jacobaraujo7** [comment]
@chunhtai I did the squash and it really worked. What needs to be done now?
I believe that this fix should be a hot-fix in the beta channel, as this makes it impossible to use the back buttons on the browser when the project is compiled with nullsafety.

File: packages/flutter/lib/src/widgets/binding.dart
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/binding.dart:668]: Hi, sorry i didn't notice this with my last review, I think the `location` should always be none null, otherwise didPushRouteInformation does not make sense.

**Review by chunhtai — COMMENTED** [review]
Sorry I didn't catch this in the last review. LGTM after this is fixed

File: packages/flutter/lib/src/widgets/binding.dart
Comment (jacobaraujo7) [reply packages/flutter/lib/src/widgets/binding.dart:668]: Ok, removed

**Pull request merged**
